### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by Daphineatim**
### In this milestone, the following were implemented;

- [ ] Created an app.rb file that will serve as the console app entry-point. It should have methods that do the following:

- List all books.
- List all people.

- Create a person (teacher or student, not a plain Person).

-  Create a book.

-  Create a rental.

- List all rentals for a given person id.

- [ ]  Defined the entry point in main.rb, this will be a method called main that is invoked at the end of the file. This method should do the following:

- Present the user with a list of options to perform.

- Lets users choose an option. (By choosing  a number)

- Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    puts 'List of Rentals: '
+    if @rentals.empty?
+      puts 'There is no rental in the list. Please add a rental!'
+    else
+      @rentals.each do |rental|
+        puts "Date: #{rental.date}. Book: '#{rental.book.title}' by #{rental.book.author}" if rental.person.id == id
+      end
+    end
+    puts "\n"
+  end
```
Comment (thecodechaser) [inline app.rb]: - [x] When there's no person(student or teacher) created and If I try to list the rentals by selecting option 6, I get the below error. Please fix this issue. (The application should be ready to handle any user interaction and should not break).
  - To fix this issue, Replace line `114` with `return if @persons.empty?`.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Daphineatim👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (Daphineatim) [reply app.rb]: Thank you for the feedback, I have worked on the issue.

**Review by roshan-bajgain — APPROVED** [review]
Hi @Daphineatim, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- PR has a good title and summary. ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**